### PR TITLE
Add me.org met site configurations

### PR DESCRIPTION
**Pull request by SeanBryan51**
These changes allow users to run a set of standard tests (experiments) for CABLE from which model output can be piped into [modelevaluation.org](modelevaluation.org). In particular we
- no longer default to running CABLE with all met files in the FLUXNET dataset.
- now use preset experiments that enforce which met sites are to be used. These are defined in the NRI Land testing workspace on [modelevaluation.org](modelevaluation.org).

This addresses issues #10 and #25.

### CODE REVIEW

File: benchcab/internal.py [hunk 1]
```diff
@@ -82,6 +82,88 @@
...
+
+    The `experiment` argument either specifies a key in `MEORG_EXPERIMENTS` or a site id
+    within the five-site-test experiment.
+
+    Assume all site ids map uniquely to a met file in MET_DIR.
+    '''
+
+    if experiment in MEORG_EXPERIMENTS["five-site-test"]:
+        # the user is specifying a single met site
+        return list(map(lambda path: path.name, MET_DIR.glob(f"{experiment}*")))
```
Comment (ccarouge) [inline benchcab/internal.py:159]:
```suggestion
        # Use next() to return the value from the `glob()` generator
        return [next(MET_DIR.glob(f"{experiment}*")).name]
```
I wonder if this syntax would be simpler. Not sure the syntax is 100% correct with the `)` so be careful if accepting on GitHub.

File: benchcab/internal.py [hunk 2]
```diff
@@ -82,6 +82,88 @@
...
+
+    met_sites = []
+    for site_id in MEORG_EXPERIMENTS[experiment]:
+        met_sites += list(map(lambda path: path.name, MET_DIR.glob(f"{site_id}*")))
```
Comment (ccarouge) [inline benchcab/internal.py:163]:
```suggestion
    met_sites = [ next(MET_DIR.glob(f"{site_id}*"))).name for site_id in MEORG_EXPERIMENTS[experiment] ]
```
Using the same syntax as above.

Another solution: if users give 1 site, when reading the config we create the experiment key "one-site" and give the site id as a value. This way we avoid the distinction between cases here.

File: benchcab/task.py [hunk 3]
```diff
@@ -151,11 +149,8 @@ def get_fluxnet_tasks(config: dict, science_config: dict) -> list[Task]:
             sci_conf_key=key,
             sci_config=science_config[key]
         )
-        for id, branch in config["realisations"].items() for site in met_sites for key in science_config
+        for id, branch in config["realisations"].items()
```
Comment (ccarouge) [inline benchcab/task.py:152]: Should the argument be `realisations` for mimicking the other arguments? Then call it with `realisations=config["realisations"]`. This way we don't have to give the whole config dictionary to this function when we only need one element of it.

I realise this might be better in a separate PR as it's different from met_sites.

**Review by ccarouge — CHANGES_REQUESTED** [review]

File: benchcab/internal.py [hunk 4]
```diff
@@ -82,6 +82,88 @@
...
+        "US-FPe",
+        "US-SRM",
+        "US-SRG",
+        "US-Ton",
+        "US-UMB",
+        "US-Var",
+        "US-Whs",
+        "US-Wkg",
+    ]
+}
```
Comment (ccarouge) [inline benchcab/internal.py:145]: A better idea to avoid distinctions between 1 site vs. other cases in the code:

```
for site_id in MEORG_EXPERIMENTS["five-site-test"]:
    MEORG_EXPERIMENTS[site_id] = [ site_id ]
```
That is adding each individual site id as an experiment id with a single value

Comment (SeanBryan51) [reply benchcab/internal.py:145]: I think we should try to avoid writing for loops in a global namespace (see this [stackoverflow post](https://stackoverflow.com/questions/3611760/scoping-in-python-for-loops)), for example:
```python
for site_id in MEORG_EXPERIMENTS["five-site-test"]:
    MEORG_EXPERIMENTS[site_id] = [ site_id ]
```
the above for loop will make `site_id` accessible in the global scope.

I personally don't mind if the code is a bit more verbose since it makes it clear that specifying a single site must be a subset of the five sites. It also avoids duplication of data in `MEORG_EXPERIMENTS`

File: benchcab/task.py (hunk 3 above)
Comment (SeanBryan51) [reply benchcab/task.py:152]: Yes I agree, we should avoid passing in the whole config dictionary if we can

File: benchcab/internal.py (hunk 2 above)
Comment (SeanBryan51) [reply benchcab/internal.py:163]: I'm happy to use `next()`, it might seem more readable than using `map()` and `lambda`

Comment (SeanBryan51) [reply benchcab/internal.py]: Here are the changes: [ce4f1b9](https://github.com/CABLE-LSM/benchcab/pull/27/commits/ce4f1b9746af00fd361680afdb75a05a9fdc1808)

File: benchcab/internal.py (hunk 1 above)
Comment (SeanBryan51) [reply benchcab/internal.py]: Here are the changes: [ce4f1b9](https://github.com/CABLE-LSM/benchcab/pull/27/commits/ce4f1b9746af00fd361680afdb75a05a9fdc1808)

**Review by ccarouge — APPROVED** [review]
Ready to merge when you want.
